metrics/web/lib/auth/data/model/user_profile_data.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:meta/meta.dart';
+import 'package:metrics/auth/domain/entities/theme_type.dart';
+import 'package:metrics/auth/domain/entities/user_profile.dart';
+import 'package:metrics_core/metrics_core.dart';
+
+/// A [DataModel] that represents the [UserProfile] entity.
+class UserProfileData extends UserProfile implements DataModel {
+  /// Creates the [UserProfileData] with the given [id] and [selectedTheme].
+  UserProfileData({
+    @required String id,
+    @required ThemeType selectedTheme,
+  }) : super(id: id, selectedTheme: selectedTheme);
+
+  /// Creates the [UserProfileData] using the [json] and the [documentId].
+  ///
+  /// Returns `null` if the given [json] is `null`.
+  factory UserProfileData.fromJson(
+    Map<String, dynamic> json,
+    String documentId,
+  ) {
+    if (json == null) return null;
+
+    return UserProfileData(
+      id: documentId,
+      selectedTheme: json['selectedTheme'] as ThemeType,
```

Comment:
Since we can't store the `enum` values in the database directly, we are storing them as a string representation, so we should map it here from `String` to `enum` value. You can find an example of mapping the enum in the [build_data_deserializer.dart](https://github.com/platform-platform/monorepo/blob/master/metrics/web/lib/dashboard/data/deserializer/build_data_deserializer.dart) file.